### PR TITLE
Feature/166 vote referendum

### LINKED ISSUE #166: Milestone 2 | Push Notifications | Vote for Referendum
* Trigger: When a new referendum occurs 
* UI : No image or icon, only the app logo
* Copy:
          - Header: "Time to vote !"
          - Body: "A New Referendum has begun, check out the proposal "
* Landing screen: (future) Referendum screen 
* Timing: (For now) Right after a new referendum has begun
Should be able to turned it off on the general PN permission screen

### PR DESCRIPTION
**Pull request by ahoseinian**
### Description
- List active referendum on the new page
- Listen to pushes on `democracy.Started` and redirect to referendum page
- Extract proposal component to be used both in Motions and Referenda screen

Fixes #166

### CODE REVIEW

File: src/screen/NotificationSettingsScreen.tsx [hunk 1]
```diff
@@ -93,14 +94,15 @@ function useTopics() {
...
+        queryClient.setQueryData<string[]>('selected_push_topics', (previousData) => {
+          if (!previousData) {
             return [];
           }
-          return subscribe ? [...data, id] : data.filter((t) => t !== id);
+          return subscribe ? [...previousData, id] : previousData.filter((t) => t !== id);
         });
         return {previousTopics};
       },
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (silva-fj) [inline src/screen/NotificationSettingsScreen.tsx]: why not using the correct type?

File: src/screen/ReferendaScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import {BN_ONE} from '@polkadot/util';
+import {useBlockTime} from 'src/hook/useBlockTime';
+
+export function ReferendaScreen() {
+  const {data, isLoading, refetch} = useReferendums();
+
+  return (
+    <Layout style={globalStyles.flex}>
+      <SafeView edges={noTopEdges}>
+        <View style={styles.container}>
```
Comment (silva-fj) [inline src/screen/ReferendaScreen.tsx]: This `View` can be removed if no styles are needed

File: src/presentational/SectionTeaserContainer.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import globalStyles, {hitSlop, standardPadding} from 'src/styles';
 type PropTypes = {
   title: string;
   onMorePress: () => void;
-  children: React.ReactNode;
+  children?: React.ReactNode;
```
Comment (silva-fj) [inline src/presentational/SectionTeaserContainer.tsx]: Why making this optional?

File: src/presentational/Proposal.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import {StyleSheet, TouchableOpacity, View} from 'react-native';
+import globalStyles, {standardPadding} from 'src/styles';
+import Padder from 'presentational/Padder';
+import {CallInspector, formatCallMeta} from 'src/packages/call_inspector/CallInspector';
+import type {Proposal as ProposalType} from '@polkadot/types/interfaces';
+
+export function Proposal({
+  proposal,
+  accessoryLeft,
+  accessoryRight,
```
Comment (silva-fj) [inline src/presentational/Proposal.tsx]: What does `accessoryLeft` and `accessoryRight` mean in a Proposal?

File: src/screen/Council/MotionsScreen.tsx [hunk 5]
```diff
@@ -107,67 +101,48 @@ function Motion({item}: {item: DeriveCollectiveProposal}) {
...
-          )}`}</Text>
-          <Padder scale={1} />
-          <CallInspector call={proposal} />
-        </View>
-      ) : null}
-    </View>
+    <Proposal
+      proposal={proposal}
+      accessoryLeft={() => {
+        return <Text category={'h4'}>{formatNumber(votes?.index)}</Text>;
```
Comment (silva-fj) [inline src/screen/Council/MotionsScreen.tsx]: Will this be different in other context?

File: src/screen/Council/MotionsScreen.tsx [hunk 6]
```diff
@@ -107,67 +101,48 @@ function Motion({item}: {item: DeriveCollectiveProposal}) {
...
+      }}
+      accessoryRight={() => {
```
Comment (silva-fj) [inline src/screen/Council/MotionsScreen.tsx]: Will this change in other context or are we always displaying those buttons when a proposal is closeable or voteable?

File: src/screen/NotificationSettingsScreen.tsx (hunk 1 above)
Comment (ahoseinian) [reply src/screen/NotificationSettingsScreen.tsx]: initially I couldn't get it to work, now after second try I managed 06c4369

File: src/presentational/SectionTeaserContainer.tsx (hunk 3 above)
Comment (ahoseinian) [reply src/presentational/SectionTeaserContainer.tsx]: for referenda I didn't add anything as body yet just made it a section header so we can navigate to.

File: src/presentational/Proposal.tsx (hunk 4 above)
Comment (ahoseinian) [reply src/presentational/Proposal.tsx]: I used the same terminology as [ui-kitten](https://akveo.github.io/react-native-ui-kitten/docs/components/button/api#button) for injecting elements on left and right of components. 

the green parts are accessories
Simulator Screen Shot - iPhone 12 - 2021-07-09 at 16 00 59

File: src/screen/Council/MotionsScreen.tsx (hunk 5 above)
Comment (ahoseinian) [reply src/screen/Council/MotionsScreen.tsx]: yes we are using Proposal also in ReferendaScreen. there we are showing the index of the proposal as accessoryLeft.

File: src/screen/Council/MotionsScreen.tsx (hunk 6 above)
Comment (ahoseinian) [reply src/screen/Council/MotionsScreen.tsx]: on Referenda we wouldn't show this buttons just the remaining time left for the referendum and voting happens on the referendum detail page.

File: src/screen/ReferendaScreen.tsx (hunk 2 above)
Comment (ahoseinian) [reply src/screen/ReferendaScreen.tsx]: done ✅

File: src/presentational/SectionTeaserContainer.tsx (hunk 3 above)
Comment (silva-fj) [reply src/presentational/SectionTeaserContainer.tsx]: I see, but as this component is a "container" the children should be mandatory.. You can pass a `Text` with and empty string as body or something like this until we have something to show or export the `Header` if it's the only thing we need.

File: src/presentational/Proposal.tsx (hunk 4 above)
Comment (silva-fj) [reply src/presentational/Proposal.tsx]: Yeah I got the idea.. My point was that those names are good for generic components like a `Button`, but `Proposal` is specific to our domain, if you don't know what a `Proposal` looks like it's hard to know what you can pass to those props. Would the `ListItem` component from UI Kitten fit the generic use case?

File: src/presentational/SectionTeaserContainer.tsx (hunk 3 above)
Comment (ahoseinian) [reply src/presentational/SectionTeaserContainer.tsx]: fdf228d
done ✅

File: src/presentational/Proposal.tsx (hunk 4 above)
Comment (ahoseinian) [reply src/presentational/Proposal.tsx]: `Proposal` can be expanded on click to show some info underneath. this is the only difference to `ListItem`. I will extract that part as a separate `ProposalInfo` component and then replace Proposal with `ListItem`.

Comment (ahoseinian) [reply src/presentational/Proposal.tsx]: 88c1763
done ✅

File: src/hook/useReferendums.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import {useApi} from 'context/ChainApiContext';
+import {useQuery} from 'react-query';
+
+export function useReferendums() {
```
Comment (silva-fj) [inline src/hook/useReferendums.ts:4]: awesome 👍🏼

File: src/layout/ReferendaSummaryTeaser.tsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import SeactionTeaserContainer from 'presentational/SectionTeaserContainer';
```
Comment (silva-fj) [inline src/layout/ReferendaSummaryTeaser.tsx]:
```suggestion
import SectionTeaserContainer from 'presentational/SectionTeaserContainer';
```

File: src/layout/ReferendaSummaryTeaser.tsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+import React from 'react';
+import {View} from 'react-native';
+
+type Props = {
+  onMorePress: () => void;
+};
+
+export function ReferendaSummaryTeaser(props: Props) {
+  return (
+    <SeactionTeaserContainer onMorePress={props.onMorePress} title="Refernda">
```
Comment (silva-fj) [inline src/layout/ReferendaSummaryTeaser.tsx]:
```suggestion
    <SectionTeaserContainer onMorePress={props.onMorePress} title="Referenda">
```

**Review by silva-fj — DISMISSED** [review]

**Review by silva-fj — APPROVED** [review]

**Pull request merged**